tests/functional/secret/test_scan_prereceive.py

```diff
@@ -100,3 +103,44 @@ def test_scan_prereceive_push_force(tmp_path: Path) -> None:
     # THEN the push is accepted because the commit containing the secret has not been
     # scanned
     local_repo.push("--force")
+
+
+def test_scan_prereceive_timeout(tmp_path: Path, slow_gitguardian_api: str) -> None:
+    # GIVEN a remote repository
+    remote_repo = Repository.create(tmp_path / "remote", bare=True)
+
+    # AND a local clone
+    local_repo = Repository.clone(remote_repo.path, tmp_path / "local")
+
+    # AND ggshield installed as a pre-receive hook
+    hook_path = remote_repo.path / "hooks" / "pre-receive"
+    hook_path.write_text(HOOK_CONTENT)
+    hook_path.chmod(0o755)
+
+    # AND a secret committed
+    secret_file = local_repo.path / "secret.conf"
+    secret_content = f"password = {GG_VALID_TOKEN}"
+    secret_file.write_text(secret_content)
+    local_repo.add("secret.conf")
+    local_repo.create_commit()
+
+    # Prepare to mock process.run to access the stderr of the push
+    stderr = b""
+    subprocess_run = subprocess.run
+
+    def run_and_copy_stderr(*args, **kwargs):
+        process = subprocess_run(*args, **kwargs)
+        nonlocal stderr
+        stderr = process.stderr
+        return process
```

Comment:
That's some nifty code :) but have you tried using pytest [capsys](https://docs.pytest.org/en/7.3.x/reference/reference.html#pytest.capture.capsys) to capture stderr? It would be less intrusive: as much as possible, functional tests should run the app without modifying it.

Reply:
I have edited `git_shell` to `log.debug` the stderr of the command. It simplifies the test, because we can use `caplog`. It will also make debugging easier and won't affect users unless they use `--debug`. WDYT ?

Reply:
Good idea 👍🏻.